### PR TITLE
Updating shared libraries to make a pip installable package.

### PR DESCRIPTION
**Pull request by bhazuka**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-205: Convert shared libraries to a python package to be installed via pip](https://bugs.earthdata.nasa.gov/browse/ORCA-205)

## Changes

* Created `tasks/shared_libraries/orca_shared` directory and `tasks/shared_libraries/orca_shared/__init__.py` file.
* Created `tasks/shared_libraries/orca_shared/database` directory and `tasks/shared_libraries/orca_shared/database/__init__.py` file.
* Created `tasks/shared_libraries/orca_shared/recovery` directory and `tasks/shared_libraries/orca_shared/recovery/__init__.py` file.
* Created `tasks/shared_libraries/README.md` file
* Created `tasks/shared_libraries/API.md` file
* Created `tasks/shared_libraries/requirements.txt` file
* Created `tasks/shared_libraries/setup.py` file
* Created `tasks/shared_libraries/bin/build_api.sh` file
* Created `tasks/shared_libraries/bin/run_tests.sh` file
* Copied shared_db file and test directories from `tasks/shared_libraries/database` to `tasks/shared_libraries/orca_shared/database`
* Copied shared_recovery file and test directories from `tasks/shared_libraries/recovery` to `tasks/shared_libraries/orca_shared/recovery`
* Modified lambdas using shared libraries to perform installs via pip. 

## Type of change

Reorganizing Shared libraries into a python package that can be installed via pip. This will simplify the installation and use of the shared libraries.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [X] CHANGELOG.md
    - [X] README
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
    - [X] Automated tests passing (if not fork) **note the recovery and database directories must be removed to use the run_tests script**
- [X] Any dependent changes have been merged and published in downstream modules
- [X] My code has passed security scanning
    - [X] Snyk
    - [X] git-secrets

## What is the current behavior?

Lambdas that utilize the shared libraries must manually copy the files to the local directory and to the proper location during builds and must manage the shared library requirements within the Lambdas requirements files.

## What is the expected correct/added behavior?

Lambdas can install the specified shared library using pip without knowing the underlying dependencies for the libraries needed. 

## Relevant logs and/or screenshots

Here is the shell script portion used for running tests and building lambdas needed for using shared libraries in addition to additional libraries being added to the lambdas requirements file.

```
## SHARED LIBS
echo "INFO: Copying ORCA shared libraries ..." 
if [ -d orca_shared ]; then
    rm -rf orca_shared
fi
mkdir -p build/orca_shared
let return_code=$?
check_rc $return_code "ERROR: Unable to create orca_shared directory."
touch build/orca_shared/__init__.py
let return_code=$?
check_rc $return_code "ERROR: Unable to create [orca_shared/__init__.py] file"
cp ../shared_libraries/database/shared_db.py build/orca_shared/
let return_code=$?
check_rc $return_code "ERROR: Unable to copy shared library [orca_shared/shared_db.py]"
cp ../shared_libraries/recovery/shared_recovery.py build/orca_shared/
let return_code=$?
check_rc $return_code "ERROR: Unable to copy shared library [orca_shared/shared_recovery.py]
```

## Possible fixes

Remove the code featured above from the lambdas test and build scripts and update the lambda requirement scripts to install the library via pip.

### CODE REVIEW

File: tasks/shared_libraries/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+pydoc-markdown>=4.0.0,<5.0.0
+## Additional validation libraries
+
+## Standard unit test libraries
+pytest==6.1.2
+coverage==5.3
+## Libraries needed for unit tests
+moto[sqs]==2.0.6
+psycopg2-binary==2.8.6
+## Libraries needed by application
```
Comment (AndrewDorn) [inline tasks/shared_libraries/requirements.txt:11]: Might be a silly question, but which application?

File: tasks/shared_libraries/setup.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    version_file = read("orca_shared", "__init__.py")
+    version_match = re.search(
+        r'^__version__ = [\'"]([^\'"]*)[\'"]', version_file, re.MULTILINE
+    )
+    if version_match:
+        return version_match.group(1)
+    raise RuntimeError("Unable to find version string.")
+
+
+# General installation requirements
```
Comment (AndrewDorn) [inline tasks/shared_libraries/setup.py:30]: "General", as in, required by both libraries? Why not separate this out?

**Review by AndrewDorn — COMMENTED** [review]
I like the automated api generation in particular. Will all of our shared libraries be under the same pip install?

File: tasks/shared_libraries/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+# Run the tests
+bash-4.2# bin/run_tests.sh
+```
+
+Note that Bamboo will run this same script via the `bin/run_tests.sh` script found
+in the cumulus-orca base of the repo.
+
+
+## Creating and Updating Shared Libraries
```
Comment (AndrewDorn) [inline tasks/shared_libraries/README.md:106]: Add bumped version number

Comment (bhazuka) [reply tasks/shared_libraries/README.md:106]: Added bullet to specify updating the version to specify a change was made.

File: tasks/shared_libraries/requirements.txt (hunk 1 above)
Comment (bhazuka) [reply tasks/shared_libraries/requirements.txt]: Added more comments and broke items up to better detail what requirements belong to what.

**Review by AndrewDorn — APPROVED** [review]
That resolves everything I could find.

File: tasks/shared_libraries/orca_shared/database/schemas/output.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    },
+    "required": [
+        "admin_database",
+        "admin_user",
+        "admin_user_password",
+        "app_user",
+        "app_user_password",
+        "database",
+        "host",
+        "port"
```
Comment (rizbi-hassan) [inline tasks/shared_libraries/orca_shared/database/schemas/output.json]: once we merge the DB migration changes into develop, these variable names should be changed as well.

Comment (bhazuka) [reply tasks/shared_libraries/orca_shared/database/schemas/output.json]: yes, the updated libraries will need to be merged over as part of the DB migration changes into develop.

File: tasks/shared_libraries/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+The ORCA shared libraries package contains several modules that group
+common functionality used in ORCA lambda functions. Below is a list of
+available libraries and their purpose.
+
+- [**database**](API.md#orca_shared.database) - The database library contains several functions for connecting to and working with a database.
+- [**recovery**](API.md#orca_shared.recovery) - The recovery library contains several functions used by ORCA recovery workflows.
+
+
+The following sections going into more detail on utilizing these libraries.
```
[the remark this answers is not in the document]
Comment (bhazuka) [reply tasks/shared_libraries/README.md]: fixed

**Review by rizbi-hassan — APPROVED** [review]

**Pull request merged**